test/unit/textbox.js

```diff
@@ -178,8 +178,11 @@
       assert.deepEqual(obj.styles[2], textbox.styles[2], 'styles match at line 2');
       assert.deepEqual(obj.styles[2][0], textbox.styles[2][0], 'styles match at index 0');
       assert.deepEqual(obj.styles[2][1], textbox.styles[2][1], 'styles match at index 1');
-      fabric.Textbox.fromObject(obj).then(function(obj2) {
```

Comment:
this test was causing a lot of problems
it took me a lot of time to understand what is wrong
So because we spread the input arg in `Text.fromObject` passing instance to `fromObject` pollutes the created instance with class props

I don't think we should care about it, though it is an edge case. No sense in passing a live instance to `fromObject` but if you think differently then we can safeguard it by running `cloneDeep` before spreading